<?xml version="1.0"?>
<!--
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
-->

<!--

This launch file was intended for use stand-alone use together with Gazebo, but is not updated anymore. It can be, however, a nice starting point for people that want to use Gazebo.

  -->

<launch>
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find cvg_sim_gazebo)/worlds/ardrone_testworld.world"/>
  </include>

  <!-- Spawn simulated quadrotor uav -->
  <include file="$(find cvg_sim_gazebo)/launch/spawn_quadrotor.launch" >
    <arg name="model" value="$(find cvg_sim_gazebo)/urdf/quadrotor_sensors.urdf.xacro"/> 
  </include>

  <node name="joy" pkg="joy" type="joy_node" output="screen" />

  <node name="fly_from_joystick" pkg="ram" type="fly_from_joystick">
    <param name="x_axis" value="2"/> 
    <param name="y_axis" value="1"/> 
    <param name="z_axis" value="4"/> 
    <param name="yaw_axis" value="3"/> 
    <param name="x_velocity_max" value="1"/>
    <param name="y_velocity_max" value="1"/>
    <param name="z_velocity_max" value="1"/>
    <param name="yaw_velocity_max" value="1"/>
    <param name="slow_button" value="6"/>
    <param name="reset_button" value="8"/>
    <param name="land_button" value="11"/>
    <param name="takeoff_button" value="12"/>
    <param name="toggle_control_button" value="7" />
    <param name="enable_stick_control_init" value="true" />
  </node>

  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="yocs_cmd_vel_mux" args="load yocs_cmd_vel_mux/CmdVelMuxNodelet nodelet_manager">
    <param name="yaml_cfg_file" value="$(find ram)/launch/ram.yaml"/>
  </node>
  
  <node pkg="mocap_optitrack" 
  type="mocap_node" 
  name="mocap_node" 
  respawn="false"
  launch-prefix=""
  required="true">
    <rosparam file="$(find mocap_optitrack)/config/mocap.yaml" command="load" />
  </node>

  <!--<node name="control" pkg="ram" type="controller" output="screen">
  	<param name="setpoint_x" value="0" /> 	
  	<param name="setpoint_y" value="0.6" />
  	<param name="setpoint_z" value="1" />
  	<param name="setpoint_yaw" value="0" />
  	<param name="gain_p_translational" value="0.08" />
  	<param name="gain_d_translational" value="0" />
  	<param name="gain_p_rotational" value="0.05" />
  	<param name="gain_d_rotational" value="0" />
  	<param name="simulation" value="true" />
  </node>-->
</launch>